### PR TITLE
feat(general): Implement authentication retry mechanism

### PR DESCRIPTION
**Pull request by OfekShimko**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sast|sca|secrets|serverless|terraform|general|graph|terraform_plan|terraform_json
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

*Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.*

Fixes # (issue)

### Description
*Include a description of what makes it a violation and any relevant external links.*

### Fix
*How does someone fix the issue in code and/or in runtime?*

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: checkov/common/bridgecrew/platform_integration.py [hunk 1]
```diff
@@ -249,6 +249,30 @@ def is_bc_token(token: str | None) -> TypeGuard[str]:
...
+    def fetch_auth_token(self, username: str, password: str) -> str:
+        retries = int(os.getenv('REQUEST_MAX_TRIES', 3))
+        request: Any = None
+        for i in range(retries):
+            request = self.http.request("POST", f"{self.prisma_api_url}/login",  # type:ignore[union-attr]
+                                        body=json.dumps({"username": username, "password": password}),
+                                        headers=merge_dicts({"Content-Type": "application/json"},
+                                                            get_user_agent_header()))
+
+            if request.status == 200:
```
Comment (Eliran-Turgeman) [inline checkov/common/bridgecrew/platform_integration.py:266]: we can use `request.ok` instead so we are not tied directly to 200, if there are any future changes

File: checkov/common/bridgecrew/platform_integration.py [hunk 2]
```diff
@@ -249,6 +249,30 @@ def is_bc_token(token: str | None) -> TypeGuard[str]:
...
+                token: str = json.loads(request.data.decode("utf8"))['token']
+                return token
+            elif request.status in [401, 403]:
```
Comment (Eliran-Turgeman) [inline checkov/common/bridgecrew/platform_integration.py:269]: I think 500 fits here too

**Review by Eliran-Turgeman — APPROVED** [review]

File: checkov/common/bridgecrew/platform_integration.py (hunk 2 above)
Comment (OfekShimko) [reply checkov/common/bridgecrew/platform_integration.py:269]: @mikeurbanski1 WDYT?

File: checkov/common/bridgecrew/platform_integration.py (hunk 1 above)
Comment (OfekShimko) [reply checkov/common/bridgecrew/platform_integration.py:266]: Are you sure that this attribute exists?

File: checkov/common/bridgecrew/platform_integration.py (hunk 2 above)
Comment (mikeurbanski1) [reply checkov/common/bridgecrew/platform_integration.py:269]: Yeah, we should retry any 5xx error

File: checkov/common/bridgecrew/platform_integration.py [hunk 3]
```diff
@@ -249,6 +249,30 @@ def is_bc_token(token: str | None) -> TypeGuard[str]:
...
+                self.raise_bridgecrew_auth_error(request.status, request.data)
```
Comment (mikeurbanski1) [inline checkov/common/bridgecrew/platform_integration.py:270]: Doesn't raising an error here prevent the retry from occurring?

File: checkov/common/bridgecrew/platform_integration.py (hunk 2 above)
Comment (OfekShimko) [reply checkov/common/bridgecrew/platform_integration.py:269]: In this if, we raise exception :) so need to to add it here

**Review by mikeurbanski1 — APPROVED** [review]

**Pull request closed without merging**